receiver/dockerstatsreceiver/receiver_test.go

```diff
@@ -149,26 +151,44 @@ func TestScrapeV2(t *testing.T) {
-		"/v1.22/containers/" + containerIDs[0] + "/json":  filepath.Join(mockFolder, "single_container", "container.json"),
-		"/v1.22/containers/" + containerIDs[0] + "/stats": filepath.Join(mockFolder, "single_container", "stats.json"),
+		"/v1.23/containers/json":                          filepath.Join(mockFolder, "single_container", "containers.json"),
+		"/v1.23/containers/" + containerIDs[0] + "/json":  filepath.Join(mockFolder, "single_container", "container.json"),
+		"/v1.23/containers/" + containerIDs[0] + "/stats": filepath.Join(mockFolder, "single_container", "stats.json"),
 	})
 	assert.NoError(t, err)
 	defer singleContainerEngineMock.Close()
 
 	twoContainerEngineMock, err := dockerMockServer(&map[string]string{
-		"/v1.22/containers/json":                          filepath.Join(mockFolder, "two_containers", "containers.json"),
-		"/v1.22/containers/" + containerIDs[1] + "/json":  filepath.Join(mockFolder, "two_containers", "container1.json"),
-		"/v1.22/containers/" + containerIDs[2] + "/json":  filepath.Join(mockFolder, "two_containers", "container2.json"),
-		"/v1.22/containers/" + containerIDs[1] + "/stats": filepath.Join(mockFolder, "two_containers", "stats1.json"),
-		"/v1.22/containers/" + containerIDs[2] + "/stats": filepath.Join(mockFolder, "two_containers", "stats2.json"),
+		"/v1.23/containers/json":                          filepath.Join(mockFolder, "two_containers", "containers.json"),
+		"/v1.23/containers/" + containerIDs[1] + "/json":  filepath.Join(mockFolder, "two_containers", "container1.json"),
+		"/v1.23/containers/" + containerIDs[2] + "/json":  filepath.Join(mockFolder, "two_containers", "container2.json"),
+		"/v1.23/containers/" + containerIDs[1] + "/stats": filepath.Join(mockFolder, "two_containers", "stats1.json"),
+		"/v1.23/containers/" + containerIDs[2] + "/stats": filepath.Join(mockFolder, "two_containers", "stats2.json"),
 	})
 	assert.NoError(t, err)
 	defer twoContainerEngineMock.Close()
 
+	noPidsStatsEngineMock, err := dockerMockServer(&map[string]string{
+		"/v1.23/containers/json":                          filepath.Join(mockFolder, "no_pids_stats", "containers.json"),
+		"/v1.23/containers/" + containerIDs[0] + "/json":  filepath.Join(mockFolder, "no_pids_stats", "container.json"),
+		"/v1.23/containers/" + containerIDs[0] + "/stats": filepath.Join(mockFolder, "no_pids_stats", "stats.json"),
+	})
+	assert.NoError(t, err)
+	defer noPidsStatsEngineMock.Close()
+
+	pidsStatsMaxEngineMock, err := dockerMockServer(&map[string]string{
+		"/v1.23/containers/json":                          filepath.Join(mockFolder, "pids_stats_max", "containers.json"),
+		"/v1.23/containers/" + containerIDs[3] + "/json":  filepath.Join(mockFolder, "pids_stats_max", "container.json"),
+		"/v1.23/containers/" + containerIDs[3] + "/stats": filepath.Join(mockFolder, "pids_stats_max", "stats.json"),
+	})
+	assert.NoError(t, err)
+	defer pidsStatsMaxEngineMock.Close()
+
```

Comment:
Thanks for the suggestion! I didn't consider changing the test structure but it makes a lot of sense. I moved the mock definition inside the test case struct as suggested. Since it is in a function and the server is created _inside_ of each `t.Run` I think it is safe closing it inside as well. Please, let me know your thoughts.
Changed here: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21111/commits/74f6e9ac38eabc17a8e11774afcde2211183a05a

Reply:
Thanks for improving that, it looks good.